qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -64,33 +75,52 @@ def __init__(
         self._data: Mapping[str, complex] = {}
         if copy:
             if validate:
-                register_length = self.__class__._validate_keys(data.keys(), register_length)
+                self._validate_keys(data.keys())
             self._data = dict(data.items())
         else:
             self._data = data
-        self._register_length = register_length
 
     @property
-    def register_length(self) -> int:
+    @abstractmethod
+    def register_length(self) -> int | None:
         """Returns the register length"""
-        return self._register_length
 
-    @register_length.setter
-    def register_length(self, reg_length: int | None) -> None:
-        self._validate_keys(self._data.keys(), reg_length)
-        self._register_length = reg_length
+    # pylint: disable=unused-argument
+    def _new_instance(
```

Comment:
Should we make this abstract so as to have a sub-class implement this. Its the aspect we expect to change right?

Reply:
I guess we can do that, yes :+1: